### PR TITLE
Add re, textwrap modules

### PR DESCRIPTION
**Pull request by S-YOU**
- Apart from performance, most should work

```
cat <<EOF >hello.py | make run
import re
print dir(re)
print re.search('[a-z]', 't')
print re.search('[a-z]', '1')
EOF

['A', 'ASCII', 'BRANCH', 'DEBUG', 'DOTALL', 'I', 'IGNORECASE', 'L', 'LOCALE', 'M', 'MULTILINE', 'S', 'SUBPATTERN', 'Scanner', 'T', 'TEMPLATE', 'U', 'UNICODE', 'VERBOSE', 'X', '_MAXCACHE', '__all__', '__class__', '__delattr__', '__dict__', '__file__', '__getattribute__', '__hash__', '__init__', '__module__', '__name__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__version__', '_alphanum_bytes', '_alphanum_str', '_cache', '_cache_repl', '_compile', '_compile_repl', '_expand', '_pattern_type', '_pickle', '_subx', 'compile', 'copy_reg', 'error', 'escape', 'findall', 'finditer', 'match', 'purge', 'search', 'split', 'sre_compile', 'sre_parse', 'sub', 'subn', 'template']
<_sre.SRE_Match object at 0xc4200eac80>
None
```

### CODE REVIEW

**Comment by trotterdylan** [comment]
This is really exciting! Do you have a sense of what the main blocking issues with it are at this point?

**Comment by S-YOU** [comment]
array.array is main behind this, I think.

**Comment by S-YOU** [comment]
Less buggy now, but way slower.

```
cat <<EOF >regex.py
import re

compiled = re.compile("[a-z]+")

print "text", compiled.search('text')
print "1234", compiled.search('1234')

EMAIL_REGEX = r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)"
email_re = re.compile(EMAIL_REGEX)

for x in ["test@test.com", "test.com", "example@example.com"]:
  print x, email_re.match(x)

inputs = ["%s@%s.%s" % (chr(i % 64 + 32)*10, chr((0xff ^ i) % 64 + 32)*10, chr((10 ^ i) % 64 + 32) * 3) for i in range(10000)]
results = [x for x in inputs if email_re.match(x)]

print "input", len(inputs), "matched", len(results), "first 10", results[:10]
EOF

tools/grumpc regex.py > regex.go &&  go build regex.go

time python regex.py; time ./regex
```

Python version took 0.050s for 10k matches. I guess, C regex cache every possible way.
```
text <_sre.SRE_Match object at 0x10180ae00>
1234 None
test@test.com <_sre.SRE_Match object at 0x101835648>
test.com None
example@example.com <_sre.SRE_Match object at 0x101835648>
input 10000 matched 2184 first 10 ['2222222222@MMMMMMMMMM.888', '3333333333@LLLLLLLLLL.999', '8888888888@GGGGGGGGGG.222', '9999999999@FFFFFFFFFF.333', 'FFFFFFFFFF@9999999999.LLL', 'GGGGGGGGGG@8888888888.MMM', 'HHHHHHHHHH@7777777777.BBB', 'IIIIIIIIII@6666666666.CCC', 'KKKKKKKKKK@4444444444.AAA', 'LLLLLLLLLL@3333333333.FFF']
python regex.py  0.03s user 0.01s system 88% cpu 0.050 total
```

Compiled version in Go took 5s  for 10k matches

```
text <_sre.SRE_Match object at 0xc420151920>
1234 None
test@test.com <_sre.SRE_Match object at 0xc4203c15c0>
test.com None
example@example.com <_sre.SRE_Match object at 0xc4203ef720>
input 10000 matched 2184 first 10 ['2222222222@MMMMMMMMMM.888', '3333333333@LLLLLLLLLL.999', '8888888888@GGGGGGGGGG.222', '9999999999@FFFFFFFFFF.333', 'FFFFFFFFFF@9999999999.LLL', 'GGGGGGGGGG@8888888888.MMM', 'HHHHHHHHHH@7777777777.BBB', 'IIIIIIIIII@6666666666.CCC', 'KKKKKKKKKK@4444444444.AAA', 'LLLLLLLLLL@3333333333.FFF']
./regex  6.68s user 0.09s system 119% cpu 5.671 total
```

**Comment by trotterdylan** [comment]
As far as performance goes, don't worry about it too much at this point. We can write up some benchmarks and then work on profiling and optimizing. I'm optimistic we can make big improvements once we understand where the bottlenecks are.

**Comment by S-YOU** [comment]
> Just double checking on this one: is this the original formatting in the PyPy code?
Nope, I need to revert some.

**Comment by S-YOU** [comment]
- Restore all formatting from pypy.
- Added textwrap module (which based on re) to stdlib

Apart from performance, most should work.

File: lib/sys.py [hunk 1]
```diff
@@ -24,6 +24,8 @@
 
 goversion = Version()
 maxint = MaxInt
+maxsize = maxint
+maxunicode = 0xffff
```
Comment (trotterdylan) [inline lib/sys.py]: I think this should be unicode.MaxRune

Comment (S-YOU) [inline lib/sys.py]: I guess re implementation won't work if it is 32bit I guess. may be hard code in re, instead of here?

Comment (S-YOU) [inline lib/sys.py]: Oh, already hard-coded there, removing this....

Comment (S-YOU) [inline lib/sys.py]: Updated to define maxunicode  as unicode.MaxRune now.
Currently not in use.

Comment (S-YOU) [inline lib/sys.py]: @trotterdylan, I will check remaining tomorrow morning my time, a bit late for now...

**Comment by trotterdylan** [comment]
This is great! Thanks for getting this working. Merging.

**Pull request merged**